src/Sylius/Bundle/ApiBundle/CommandHandler/PickupCartHandler.php

```diff
@@ -45,18 +46,23 @@ final class PickupCartHandler implements MessageHandlerInterface
     /** @var RandomnessGeneratorInterface */
     private $generator;
 
+    /** @var CartVisitorsCustomerContextInterface */
+    private $cartVisitorsCustomerContext;
```

Comment:
IMO command handler shouldn't be session aware; could you tell me why do you need this context? You should operate on logged in user while checking access to cart

Reply:
As mentioned by @GSadee, handlers need to be stateless, as they may be executed in different contexts (CLI, UI, Api, tests, etc.). What have you tried to achieve?